operator/apis/platform/v1alpha1/overlay_types.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// K8sObjectOverlay is an overlay that applies a set of patches to a resource.
+//
+// ## Examples
+//
+// ### Adding an annotation to a resource
+//
+// ```
+// apiVersion: v1
+// kind: ServiceAccount
+// name: central
+// patches:
+//   - path: metadata.annotations.eks\.amazonaws\.com/role-arn
+//     value: "\"arn:aws:iam:1234:role\""
+//
+// ```
+//
+// ### Adding an environment variable to a deployment
+//
+// ```
+// apiVersion: apps/v1
+// kind: Deployment
+// name: central
+// patches:
+//   - path: spec.template.spec.containers[name:central].env[-1]
+//     value: |
+//     name: MY_ENV_VAR
+//     value: value
+//
+// ```
+//
+// ### Adding an ingress to a network policy
+//
+// ```
+// apiVersion: networking.k8s.io/v1
+// kind: NetworkPolicy
+// name: allow-ext-to-central
+// patches:
+//   - path: spec.ingress[-1]
+//     value: |
+//     ports:
+//   - port: 999
+//     protocol: TCP
+//
+// ```
+//
+// ### Changing the value of a configMap
+//
+// ```
+// apiVersion: v1
+// kind: ConfigMap
+// name: central-endpoints
+// patches:
+//   - path: data
+//     value: |
+//     endpoints.yaml: |
+//     disableDefault: false
+//
+// ```
+//
+// ### Adding a container to a deployment
+//
+// ```
+// apiVersion: apps/v1
+// kind: Deployment
+// name: central
+// patches:
+//   - path: spec.template.spec.containers[-1]
+//     value: |
+//     name: nginx
+//     image: nginx
+//     ports:
+//   - containerPort: 8000
```

Comment:
Is the indentation correct here?

Reply:
Good point. Just found out that [this](https://tip.golang.org/doc/comment#syntax) exists. Fixed the docs